marxs/source/labSource.py

```diff
@@ -3,15 +3,28 @@
 
 from .source import Source
 from ..optics.base import FlatOpticalElement
+from ..optics.polarization import polarization_vectors
 
 
-class FarLabConstantPointSource(Source, FlatOpticalElement):
+class PolarizedSource(Source):
+    def __init__(self, **kwargs):
+        super(PolarizedSource, self).__init__(**kwargs)
+    
+    def random_polarization(self, n, dir_array):
+        # randomly choose polarization (temporary)
+        angles = np.random.uniform(0, 2 * np.pi, n)
+        return polarization_vectors(dir_array, angles)
+
+
+class FarLabConstantPointSource(PolarizedSource, FlatOpticalElement):
     '''Simple in-lab source used with aperture
     
     - assumes point source is far from a rectangular aperture, and only the photons that pass through are tracked
     - photon start positions uniformly distributed within rectangular aperture (reasonable approximation if source is far)
     - photon direction determined by location of source, and selected photon starting position
     - aperture is parallel to y-z plane
+    - polarization angle is assumed relative to positive y axis unless direction is exactly along y axis,
+      in which case polarization is relative to positive x axis, ALL GLOBAL AXES
```

Comment:
This definition of poalrization if true everywhere in the code and not just here, I guess? It should probably be explained in the general `Source` then.